MonoGame.Framework/Utilities/Png/PngWriter.cs

```diff
@@ -0,0 +1,208 @@
+﻿// MonoGame - Copyright (C) The MonoGame Team
+// This file is subject to the terms and conditions defined in
+// file 'LICENSE.txt', which is part of this source code package.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using MonoGame.Utilities.ZLib;
+using Microsoft.Xna.Framework.Graphics;
+using Microsoft.Xna.Framework;
+
+namespace MonoGame.Utilities.Png
+{
+    public class PngWriter
+    {
+        private const int bitsPerSample = 8;
+        private ColorType colorType;
+        private Color[] textureData;
+
+        public PngWriter()
+        {
+            // TODO: currently ignoring alpha values while writing PNG images
+            colorType = ColorType.Rgb;
+        }
+
+        public Stream Write(Texture2D texture2D)
+        {
+            textureData = new Color[texture2D.Width * texture2D.Height];
+            texture2D.GetData<Color>(textureData);
```

Comment:
@tgjones @KonajuGames  - `Texture2D.GetData<>()` doesn't do any sort of color reformatting does it?  Like if the texture is a `Bgra4444` then a call to `Texture2D.GetData<Color>()` will result in a bad color array correct?

I suspect this will have to be more complex eventually to deal with different formats.